src/tests/ftest/control/dmg_system_reformat.py

```diff
@@ -39,25 +39,27 @@ def test_dmg_system_reformat(self):
         :avocado: tags=control,dmg,system_reformat
         :avocado: tags=DmgSystemReformatTest,test_dmg_system_reformat
         """
+        dmg = self.get_dmg_command().copy()
+
         # Create pool using 90% of the available NVMe capacity
-        pools = [add_pool(self)]
+        pools = [add_pool(self, dmg=dmg)]
 
         self.log.info("Check that new pool will fail with DER_NOSPACE")
-        self.get_dmg_command().exit_status_exception = False
+        dmg.exit_status_exception = False
         pools.append(add_pool(self, create=False, **get_size_params(pools[0])))
         try:
             pools[-1].create()
         except TestFail as error:
             self.log.info("Pool create failed: %s", str(error))
-            if "-1007" not in self.get_dmg_command().result.stderr_text:
+            if "-1007" not in dmg.result.stderr_text:
```

Comment:
Since we see this `Pool create failed:` message logged in https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-11712/2/artifact/Functional%20Hardware%20Medium/control/dmg_system_reformat.py/repeat001/job.log:
```
2023-03-15 19:42:07,976 process          L0604 INFO | Running '/usr/bin/dmg -o /etc/daos/daos_control.yml -d -j server set-logmasks'
2023-03-15 19:42:08,076 process          L0416 DEBUG| [stderr] DEBUG 19:42:08.076844 main.go:238: debug output enabled
2023-03-15 19:42:08,077 process          L0416 DEBUG| [stderr] DEBUG 19:42:08.077219 main.go:266: control config loaded from /etc/daos/daos_control.yml
2023-03-15 19:42:08,090 process          L0416 DEBUG| [stderr] DEBUG 19:42:08.090853 server.go:54: set log masks request: &{unaryRequest:{request:{timeout:0 deadline:{wall:0 ext:0 loc:<nil>} Sys: HostList:[]} rpc:<nil>} Masks:}
2023-03-15 19:42:08,091 process          L0416 DEBUG| [stderr] DEBUG 19:42:08.091187 server.go:52: DAOS set engine log masks request: sys:"daos_server-2.3.103"
2023-03-15 19:42:08,091 process          L0416 DEBUG| [stderr] DEBUG 19:42:08.091436 rpc.go:257: request hosts: [wolf-142:10001]
2023-03-15 19:42:08,160 process          L0416 DEBUG| [stderr] DEBUG 19:42:08.160677 server.go:69: DAOS set engine log masks response: &{HostErrorsResp:{HostErrors:map[]}}
2023-03-15 19:42:08,160 process          L0416 DEBUG| [stderr] DEBUG 19:42:08.160719 server.go:61: set log masks response: &{HostErrorsResp:{HostErrors:map[]}}
2023-03-15 19:42:08,160 process          L0416 DEBUG| [stdout] {
2023-03-15 19:42:08,161 process          L0416 DEBUG| [stdout]   "response": {
2023-03-15 19:42:08,161 process          L0416 DEBUG| [stdout]     "host_errors": {}
2023-03-15 19:42:08,161 process          L0416 DEBUG| [stdout]   },
2023-03-15 19:42:08,161 process          L0416 DEBUG| [stdout]   "error": null,
2023-03-15 19:42:08,161 process          L0416 DEBUG| [stdout]   "status": 0
2023-03-15 19:42:08,161 process          L0416 DEBUG| [stdout] }
2023-03-15 19:42:08,175 process          L0686 INFO | Command '/usr/bin/dmg -o /etc/daos/daos_control.yml -d -j server set-logmasks' finished with 0 after 0.12063097953796387s
2023-03-15 19:42:08,176 dmg_system_refor L0053 INFO | Pool create failed: DmgJsonCommandFailure('pool create failed: DER_NOSPACE(-1007): No space on storage target',)
```
I think we can use the `error` to look for the `-1007`:
```suggestion
            if "-1007" not in str(error):
```

Reply:
Please see the update.